src/objects/tokens/mirage.js

```diff
@@ -0,0 +1,48 @@
+const { refObject, Rotator } = require("../../wrapper/api");
+const { System, Planet } = require("../../lib/system/system");
+
+const MIRAGE_ATTRS = {
+    localeName: "planet.mirage",
+    resources: 1,
+    influence: 2,
+    trait: ["cultural"],
+    position: { x: 2, y: -1.25 },
+    legendary: true,
+    legendaryCard: "card.legendary_planet:pok/mirage_flight_academy",
+};
+let MIRAGE_SYSTEM = 0;
+
+function attachMirage(obj) {
+    const pos = obj.getPosition();
+    const systemObj = System.getSystemTileObjectByPosition(pos);
+    if (systemObj) {
+        const system = System.getBySystemTileObject(systemObj);
+        if (MIRAGE_SYSTEM) {
+            const prevMirageSystem = System.getByTileNumber(MIRAGE_SYSTEM);
+            console.log("Detaching mirage from", MIRAGE_SYSTEM);
+            // mirage can only go in 0 planet systems
+            prevMirageSystem.planets.splice(0, 1);
+            MIRAGE_SYSTEM = 0;
+        }
+        if (system.planets.length === 0) {
```

Comment:
I'd suggest moving the "abort" logic to the very top with a return in it, so it's easy to see there's no need to keep reading stuff below.  e.g.:
```
const system = ...
removeExistingMirage();
if (system.planets.length > 0) {
    Broadcast.chatAll('Mirage must be attached to a system with 0 planets');
    return;
}
... now on with attaching ...
```

Reply:
maybe there should be a new Broadcast.error() method that could also print a stack trace to the script console, etc...